src/test/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTracesUtilities.java

```diff
@@ -0,0 +1,122 @@
+package org.datadog.jenkins.plugins.datadog.traces;
+
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.jenkinsci.plugins.workflow.actions.TimingAction;
+import org.jenkinsci.plugins.workflow.cps.nodes.StepAtomNode;
+import org.jenkinsci.plugins.workflow.cps.nodes.StepEndNode;
+import org.jenkinsci.plugins.workflow.cps.nodes.StepStartNode;
+import org.jenkinsci.plugins.workflow.flow.FlowExecution;
+import org.jenkinsci.plugins.workflow.graph.FlowEndNode;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.graph.FlowStartNode;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DatadogTracesUtilities {
+
+    public static final Long SAMPLE_TIME = 999L;
+
+    public static final String SAMPLE_FLOW_START_NODE_ID = "flowStartNodeId";
+    public static final String SAMPLE_FLOW_START_NODE_NAME = "flowStartNodeName";
+    public static final String SAMPLE_FLOW_END_NODE_NAME = "flowEndNodeName";
+    public static final String SAMPLE_FLOW_END_NODE_ID = "flowEndNodeId";
+    public static final String SAMPLE_BLOCK_START_NODE_ID = "blockStartNodeId";
+    public static final String SAMPLE_BLOCK_START_NODE_NAME = "blockStartNodeName";
+
+    public static final String SAMPLE_BLOCK_END_NODE_ID = "blockEndNodeId";
+    public static final String SAMPLE_BLOCK_END_NODE_NAME = "blockEndNodeName";
+
+    public static final String SAMPLE_STEP_START_NODE_ID = "stepStartNodeId";
+    public static final String SAMPLE_STEP_START_NODE_NAME = "stepStartNodeName";
+    public static final String SAMPLE_STEP_END_NODE_ID = "stepEndNodeId";
+    public static final String SAMPLE_STEP_END_NODE_NAME = "stepEndNodeName";
+
+    public static final String SAMPLE_STEP_ATOM_NODE_ONE_ID = "stepAtomNodeOneId";
+    public static final String SAMPLE_STEP_ATOM_NODE_ONE_NAME = "stepAtomNodeOneName";
+    public static final String SAMPLE_STEP_ATOM_NODE_TWO_ID = "stepAtomNodeTwoId";
+    public static final String SAMPLE_STEP_ATOM_NODE_TWO_NAME = "stepAtomNodeTwoName";
+
+    private static Map<String, FlowNode> flowNodeById;
+
+    public static Map<String, FlowNode> getDummyPipeline() {
```

Comment:
Could you also write a test for a trace for a job/`FreestyleProject`?

Reply:
Added! 👍